File: S3 LAB/JAVA/BinarySearch.java
import java.util.*;
class BinarySearch{
	public static void main(String arg[]){
		Scanner sc = new Scanner(System.in);
		int i,j,limit,beg,mid,end,item,temp;
		
		System.out.println("Enter size of the array: ");
		limit = sc.nextInt();
		int arr[] = new int[limit];
		System.out.println("Enter "+limit+" numbers : \n");
		for(i=0;i<limit;i++){
			arr[i] = sc.nextInt();
		}
		
		for(j=limit-1;j>=0;j--){
			for(i=0;i<j;i++ ){
				if(arr[i]>arr[i+1]){
					temp = arr[i];
					arr[i] = arr[i+1];
					arr[i+1] = temp;
				}
			}
		}
		System.out.println("Sorted array: \n");
		for(i=0;i<limit;i++){
			System.out.println("\t"+arr[i]);
		}
		System.out.println("Enter item to search");
		item = sc.nextInt();
		beg = 0;
		end = limit-1;
		mid = (beg+end)/2;
		while((arr[mid] != item) && (beg<=end)){
			if(item < arr[mid]){
				end = mid-1;
				mid = (beg+end)/2;
			}
			else{
				beg = mid+1;
				mid = (beg+end)/2;
			}
		}
		if(arr[mid] == item){
			System.out.println("Element found at : "+ (mid+1));
		}
		else{
			System.out.println("Element not found\n");
		}
		
	}
}

File: S3 LAB/DS/BinarySearch.c
#include<stdio.h>

void main(){
	int arr[50],i,j,limit,num,beg=0,mid,end,flag=0,temp;
	printf("How many elements do you have: ");
	scanf("%d",&limit);
	end = limit-1;
	mid = (beg+end)/2;
	
	printf("\nEnter %d values: \n",limit);
	for(i=0;i<limit;i++){
		scanf("%d",&arr[i]);
	}
	
	for(i=limit-1;i>=0;i--){
		for(j=0;j<i;j++){
			if(arr[j]>arr[j+1]){
				temp = arr[j];
				arr[j] = arr[j+1];
				arr[j+1] = temp;
			}
		}
	}
	
	printf("Sorted array: \n");
	for(i=0;i<limit;i++){
		printf("%d \t",arr[i]);
	}
	
	printf("\nEnter the value to search: ");
	scanf("%d",&num);
	
	
	while(beg<=end){
		if(num == arr[mid]){
			printf("\n %d found at position: %d",num,mid+1);
			flag = 1;
			break;
		}	
		else if(num>arr[mid]){
			beg = mid+1;
			mid = (beg+end)/2;
		}
		else{
			end = mid-1;	
			mid = (beg+end)/2;
		}
	}
	if(flag == 0){
		printf("%d not found",num);
	}
	
}
